NLMechaFighter/Assets/Scripts/Fighters/Millinator/MillinatorPunch.cs

```diff
@@ -32,6 +39,29 @@ public override void Complete(MillinatorController runner)
         {
             base.Complete(runner);
             runner.anim.ResetTrigger(_punchAnim);
+            _donePunching = false;
+        }
+
+        private void CheckHit(MillinatorController runner)
+        {
+            Vector3 pos = runner.transform.position;
+            Vector3 rayOrigin = new Vector3(pos.x, pos.y + 2, pos.z);
+
+            if(Physics.Raycast(rayOrigin, runner.transform.TransformDirection(Vector3.forward), out _hit, 5f))
+            {
+                IDestroyable destroyable = _hit.transform.gameObject.GetComponent<IDestroyable>();
+                if (destroyable != null)
+                {
+                    destroyable.DestructionUpdate();
+                }
+
+                IFighter fighter = _hit.transform.gameObject.GetComponent<IFighter>();
+                if (fighter != null)
+                {
+                    //calculate how close you are to other fighter
+                    //deal damage approximatly on your distance
+                }
```

Comment:
Je kan hier gebruik maken van TryGetComponent in plaats van GetComponent. Het is niet nodig, maar misschien iets netter.
if (_hit.transform.gameObject.TryGetComponent(out IDestroyable destroyable)){
destroyable.DestructionUpdate();
}
Meer info hier: https://stackoverflow.com/questions/71094659/is-there-a-reason-why-i-should-use-trygetcomponent-instead-of-getcomponent

Reply:
oh goeie